components/in_memory_engine/src/engine.rs

```diff
@@ -251,7 +251,9 @@ impl RegionCacheMemoryEngineCore {
             return RegionCacheStatus::NotInCache;
         };
 
-        if region_meta.get_region().epoch_version < region.epoch_version {
+        if region_meta.get_region().epoch_version < region.epoch_version
+            || region_meta.get_region().in_flash_back
```

Comment:
`region_meta.get_region().in_flash_back` can never be true because in that case it is already evicted. You should check `region.in_flash_back` here. 
BTW, maybe you also need to check `in_flash_back` in `load_region` call.

Reply:
Yes, it should be region.in_flashback